assignment7.txt

```diff
@@ -0,0 +1,18 @@
+Olivier Gabison
+
+1. ++*p dereferences the value that the p variable is pointing to and then increments it. While *p++ increments the pointer,
+and then dereferences it. Finally, *++p first increments the pointer and then dereferences it.
+
+2. No, order of precedence is not done through left to right order.
+
+3. I think the best uses of pointers is passing by reference and dynamic starting sizes on arrays.
```

Comment:
right! they also allow references to functions and reduce complexity of programs!